File: Exhibition.Core/Models/Terminal.cs


namespace Exhibition.Core.Models
{
    public class Terminal
    {
        [Newtonsoft.Json.JsonProperty("ip")]
        public string Ip { get; set; }

        [Newtonsoft.Json.JsonProperty("name")]
        public string Name { get; set; }

        [Newtonsoft.Json.JsonProperty("description")]
        public string Description { get; set; }

        [Newtonsoft.Json.JsonProperty("schematic")]
        public string Schematic { get; set; }

        [Newtonsoft.Json.JsonProperty("endpoint")]
        public string Endpoint { get; set; }

        [Newtonsoft.Json.JsonProperty("windows")]
        public Window[] Windows { get; set; }
    }
}

File: Exhibition.Portal.Web/src/app/models/terminal.ts
export class Terminal {
    public ip: string;
    public name: string;
    public description: string;
    public schematic: string;
    public endpoint: string;
    public editable:boolean;
    public windows: Window[];
}
export class Window {
    public id: number
    public location: Point;
    public size:Size;
    public monitor:number;
}
export class Point {
    public x: number;
    public y: number;
}
export class Size {
    public width: number;
    public height: number;
}